pennylane_qiskit/qiskit_device2.py

```diff
@@ -0,0 +1,625 @@
+        if hasattr(val, "__dict__"):
+            options_dict.update(qiskit_options_to_flat_dict(val))
+        elif val is not None:
+            options_dict[key] = val
+    return options_dict
+
+
+class QiskitDevice2(Device):
+    r"""Hardware/simulator Qiskit device for PennyLane.
+
+    Args:
+        wires (int or Iterable[Number, str]]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers (i.e., ``[-1, 0, 2]``)
+            or strings (``['aux_wire', 'q1', 'q2']``).
+        backend (Backend): the initialized Qiskit backend
+
+    Keyword Args:
+        shots (int or None): number of circuit evaluations/random samples used
+            to estimate expectation values and variances of observables.
+        use_primitives (bool): whether or not to use Qiskit Primitives. Defaults to False. If True,
+            getting expectation values and variance from the backend will use a Qiskit Estimator,
+            and getting probabilities will use a Qiskit Sampler. Other measurement types will continue
+            to return results from the backend without using a Primitive.
+        options (Options): a Qiskit Options object for specifying handling the Qiskit task
+            (transpiliation, error mitigation, execution, etc). Defaults to None. See Qiskit documentation
+            for more details.
+        session (Session): a Qiskit Session to use for device execution. If none is provided, a session will
+            be created at each device execution.
+        **kwargs: transpilation and runtime kwargs to be used for measurements without Qiskit Primitives.
+            If any values are defined both in ``options`` and in the remaining ``kwargs``, the value
+            provided in ``options`` will take precedence. These kwargs will be ignored for all Primitive-based
+            measurements on the device.
+    """
+
+    operations = set(QISKIT_OPERATION_MAP.keys())
+    observables = {
+        "PauliX",
+        "PauliY",
+        "PauliZ",
+        "Identity",
+        "Hadamard",
+        "Hermitian",
+        "Projector",
+    }
+
+    # pylint:disable = too-many-arguments
+    def __init__(
+        self,
+        wires,
+        backend,
+        shots=1024,
+        use_primitives=False,
+        options=None,
+        session=None,
+        **kwargs,
+    ):
+        if shots is None:
+            warnings.warn(
+                "Expected an integer number of shots, but received shots=None. Defaulting "
+                "to 1024 shots. The analytic calculation of results is not supported on "
+                "this device. All statistics obtained from this device are estimates based "
+                "on samples.",
+                UserWarning,
+            )
+
+            shots = 1024
+
+        self.options = options or Options()
+        if self.options.execution.shots == 4000:  ## 4000 is default value in Qiskit.
```

Comment:
I thought we were gonna leave the default at 1024 for consistency across the devices, which makes sense to me. But changing to 4000 also makes sense to avoid ambiguity. Either way I think this can be addressed in a separate PR. Maybe just changing to 4000 here makes sense though.

Reply:
I'm not set on it either way, they're both a little inconvenient 🤷‍♀️

Reply:
I'll leave it as is for now.